tests/unit/s2n_drbg_test.c

```diff
@@ -144,87 +144,274 @@ const char nist_reference_returned_bits_hex[] =
+"cdecc083b5c4bd38b573c5e109fc1cbf2dbc92982ed6d24cf4d2cb06383ba27d51dc064a8cf42205420d1c62505b987c774502b93e837485284a137cf2380872"
+"ab1478cc75bbfdc930fefeff1e6f8e7b822213976fc556ee7f71550d128620766d8353b3c54c59934e7fa57e8b857c7daef8e9d82705fd1fec7b19d5c3b5af4f"
+"f64d6ac3fd9beda9ea4b3fc2921e967d126ab4e747eb4f29bfd949f94119271052ff52501cb45af6bf9499ef5926423f61bd4dac0b28fc95aa886c8226ea27b4"
+"45e3bfafaa00941459b373aa00e6096b7527110f4c6e4e7d6661829dfe4ad83ab23a5779dc7f8116b73b929fc3da43ad346a3e11364e1b453e66335b5cbf59b1"
+"8bc98d27050e713d19551eadae19d14f3b1304aa6f736d9bec6c7cc90c85412a4e6a7218b8a7ce46b1cb93fa1b296eee789aae0cb2a58fe7a80e26f0a1018dc8"
+"794128d681d355336148eeb4dd882d76ef984f2a8ceb4b451e9b37beac0fb50247bd595ede2de000f4397988d3f52e0b3b293208c16b5f45d6032bc5ef20b3c8"
+"3571383b971291e37616f1c6af623a1d576cc5f95090563654420f184fc6663242aeffb06052a29e84380db38d9f3ed09823dfef1774c14120fb010600b7b274"
+"b7723a01d35011a9a8e0e82a8ade37d5e992d8cc5390e2409e185a1bcede39b742fa1fa541572aa36b16be410f3ca0ca853ffd7cea2996e852b888fe9b1263f9"
+"0d42ad75a67008674be7d81bb2ca9f0625b55ea37e1985d3a27cac38e347824ff03caaaeae9b646a220a8a672cb28ba869d0f53e3317be00128fa965a8ea7b39"
+"d5be1698e011122874ea19c562859e552c68d2734a39f72d6f8eca8ce30916504bf224670f9a7b6afb57830a28fec0fe818df449f5e6eb8b49a3397df4bd3506"
+"c834a90e72a5ec0148942c9fbde8b90f058866ac3ff8cc854bda3d33f6072bc0e5a22c7d0cd801f091276ed7b7e3d6495df9546b9fc45e46aa6b89d3dda207cc"
+"6553c630a07ddf7dc2110c3aa3c5ac0c488a0345e07571cd71df115ba37ea4676935be72a6033aeca7ac6fcce5654dae38f5777b5cff34b156539b42ed6dc93c"
+"ed703d9273bb9462ac400ee8d587ea3c4d6c27aa014defcb6ca6fe885272bcb4b6ba0822f42941071bf635b41d997c631b680d91b23ee48351041dc274900821";
+
 
 /* This function over-rides the s2n internal copy of the same function */
-int nist_fake_urandom_data(struct s2n_blob *blob)
+int nist_fake_128_urandom_data(struct s2n_blob *blob)
 {
     /* At first, we use entropy data provided by the NIST test vectors */
-    GUARD(s2n_stuffer_read(&nist_reference_entropy, blob));
+    GUARD(s2n_stuffer_read(&nist_aes128_reference_entropy, blob));
 
     return 0;
 }
 
-int main(int argc, char **argv)
+int nist_fake_256_urandom_data(struct s2n_blob *blob)
 {
-    uint8_t data[256] = {0};
-    struct s2n_drbg drbg = {0};
-    struct s2n_blob blob = {.data = data, .size = 64 };
-    struct s2n_timer timer;
-    uint64_t drbg_nanoseconds;
-    uint64_t urandom_nanoseconds;
-    struct s2n_stuffer nist_reference_personalization_strings;
-    struct s2n_stuffer nist_reference_returned_bits;
-    struct s2n_stuffer nist_reference_values;
-    struct s2n_config *config;
+    /* At first, we use entropy data provided by the NIST test vectors */
+    GUARD(s2n_stuffer_read(&nist_aes256_reference_entropy, blob));
 
-    BEGIN_TEST();
+    return 0;
+}
 
-    EXPECT_NOT_NULL(config = s2n_config_new())
+int main(int argc, char **argv)
```

Comment:
I did not check the entire Test code, because I assume that the AES128 test code was not changed and that the AES256 test code is a copy paste of it.
1) If I am wrong and the AES256 test is not exactly the same as the AES128 test code please let me know.
2) If it is then the only difference is the arrays used for each test. Please merge the code into a function that receives the relevant arrays as inputs.

Reply:
1. It is the same
2. Refactored out a shared base